homeassistant/components/smartthings/climate.py

```diff
@@ -344,18 +347,32 @@ def __init__(self, device):
         # the entity state ahead of receiving the confirming push updates
         self.async_schedule_update_ha_state()
 
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self._device.switch_on(set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self._device.switch_off(set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
     async def async_update(self):
         """Update the calculated fields of the AC."""
-        operations = set()
+        modes = {HVAC_MODE_OFF}
```

Comment:
This is fine as it was already like this, but it has the downside that it makes the order of the controls random, and that order is also used in the UI I think (probably shouldn't).

Reply:
It has the effect of showing it alphabetically, I believe.  Yes, I the UI should order them as it intends instead of relying on each platform to return a specific sequence. 👍